sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaPruning.scala

```diff
@@ -40,11 +40,19 @@ private[sql] object ParquetSchemaPruning extends Rule[LogicalPlan] {
       plan
     }
 
+  // `PhysicalOperation` pattern returns relation operator's outputs if there is no
+  // projects on it. In this case, we don't need to do schema pruning.
+  private def directOutput(projects: Seq[NamedExpression], plan: LogicalPlan): Boolean = {
```

Comment:
@viirya, just for clarification, does this relate to the fix?

Reply:
Thanks for asking. @HyukjinKwon 

This check is useful during preparing this fix, but can be removed from this fix now. The added tests can pass without this.

Reply:
My second thought is, maybe we can keep this.

The reason is we can skip unnecessary transforming. Now this rule applies on filter and underlying relation too. It is unnecessary.

Reply:
Since it is actually not related to this fix, if you and others are agreed to keep it, I can make a small PR for it.